### ISSUE TITLE
Running `vagrant up` twice (on two `Vagrantfiles`) causes one of the VMs to loose connectivity

### ISSUE THREAD

**Issue by rustydb** [vagrant-libvirt/vagrant-libvirt#1535]
**Describe the bug**

I am building multiple boxes via separate `Vagrantfiles`, when `vagrant up` is ran against both files one of them will fail to boot. I am using multiple `Vagrantfiles` because I am using `packer` and that's how it works, it makes a `Vagrantfile` for each build.

If I watch the output of the libvirt deamon (`libvirtd`) and look at `net-dhcp-leases` I see that both machines get an IP lease but the first lease vanishes when the 2nd lease comes in.

Output of first box:
```bash
..==> output: Creating shared folders metadata...
==> output: Starting domain.
/==> output: Waiting for domain to get an IP address...
==> output: Waiting for machine to boot. This may take a few minutes...
    output: SSH address: 192.168.121.203:22
    output: SSH username: root
    output: SSH auth method: password
    output: Warning: Connection refused. Retrying...
    output: Warning: Connection refused. Retrying...
==> output: Removing domain...
==> output: Deleting the machine folder
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'running'. The machine is in the
'inaccessible' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

The primary issue for this error is that the provider you're using
is not properly configured. This is very rarely a Vagrant issue.
```

Output of second box:
```bash
==> output: Creating shared folders metadata...
==> output: Starting domain.
==> output: Waiting for domain to get an IP address...
==> output: Waiting for machine to boot. This may take a few minutes...
    output: SSH address: 192.168.121.220:22
    output: SSH username: root
    output: SSH auth method: password
    output: Warning: Connection refused. Retrying...
    output:
    output: Inserting generated public key within guest...
    output: Removing insecure key from the guest if it's present...
    output: Key inserted! Disconnecting and reconnecting using new SSH key...
==> output: Machine booted and ready!
```

The output of `net-dhcp-leases` is:
```
 Expiry Time           MAC address         Protocol   IP address           Hostname   Client ID or DUID
------------------------------------------------------------------------------------------------------------------------------------------------
 2022-08-08 19:28:50   52:54:00:0e:85:85   ipv4       192.168.121.220/24   install    ff:00:12:34:56:00:01:00:01:2a:82:d6:e0:52:54:00:12:34:56
```

When I run `watch` on `net-dhcp-leases` I see the first lease come in and then disappear.

If all my boxes use the same network, running two at the same time fails. If all my boxes use different networks, running two at the same time succeeds the _first time_.

I have 5 builds that build in a "layered" manner:
- Layer 1: A base box (sles15-base)
- Layer 2: Two derivative boxes from the base (pit-common and ncn-common, both build from sles15-base.)
- Layer 3: Two derivative boxes from one of the previous derivatives (kubernetes & storage-ceph, both build from ncn-common)

Currently this pipeline is packer based, and it builds Googlecompute images and bare-metal images via qemu. I am adding a Vagrant builder so we can also build vagrant boxes, but without running the packer provisioners for qemu or google against them. I'm just having an issue with Vagrant and libvirt.

If I use the same network for all my builds, then either pit-common or ncn-common fails. I can't seem to run two builds at the same time, period. Both machines get an IP lease, but one looses its lease when the 2nd machine receives its lease.

Lastly, if I build each box *one at a time* then it works.. but that's really slow. It's also not how the Jenkins pipeline is setup, so if I had to build them one at a time then I'd have to special case the Vagrant packer builds.

If I use different networks for all of my builds, then pit-common and ncn-common both get leases in their own networks and build fine. However, kubernetes or storage-ceph hit a new issue where one of them fails to even send a DHCPREQUEST (despite having their own networks).

When I say different networks, I mean each box receives its own network defined by the boxname:

```xml
<network>
  <name>ncn-common.box</name>
  <forward mode='nat'/>
  <bridge name='virbr1' stp='on' delay='0'/>
  <ip address='192.168.123.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.123.2' end='192.168.123.254'/>
    </dhcp>
  </ip>
</network>
<network>
  <name>pit-common.box</name>
  <forward mode='nat'/>
  <bridge name='virbr2' stp='on' delay='0'/>
  <ip address='192.168.124.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.124.2' end='192.168.124.254'/>
    </dhcp>
  </ip>
</network>
<network>
  <name>kubernetes.box</name>
  <forward mode='nat'/>
  <bridge name='virbr3' stp='on' delay='0'/>
  <ip address='192.168.125.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.125.2' end='192.168.125.254'/>
    </dhcp>
  </ip>
</network>
<network>
  <name>storage-ceph.box</name>
  <forward mode='nat'/>
  <bridge name='virbr4' stp='on' delay='0'/>
  <ip address='192.168.126.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.126.2' end='192.168.126.254'/>
    </dhcp>
  </ip>
</network>
```

On each one, I ran:
```bash
sudo virsh net-define
sudo virsh net-autostart
sudo virsh net-start
```
So each of the networks are up, active, and running in their own subnets. This wasn't ideal, but it let two of my builds run. However something is still off because when builds 3 and 4 run one of them fails to send a lease.

When I start builds 3 and 4 (kubernetes and storage-ceph), I only see DHCPREQUESTs for one or the other.
```
dnsmasq-dhcp[23791]: DHCPREQUEST(virbr4) 192.168.123.140 52:54:00:1e:69:36
dnsmasq-dhcp[23791]: DHCPNAK(virbr4) 192.168.123.140 52:54:00:1e:69:36 wrong network
dnsmasq-dhcp[23791]: DHCPDISCOVER(virbr4) 52:54:00:1e:69:36
dnsmasq-dhcp[23791]: DHCPOFFER(virbr4) 192.168.126.75 52:54:00:1e:69:36
dnsmasq-dhcp[23791]: DHCPREQUEST(virbr4) 192.168.126.75 52:54:00:1e:69:36
dnsmasq-dhcp[23791]: DHCPACK(virbr4) 192.168.126.75 52:54:00:1e:69:36 ncn
```

On the other hand, when I start builds 1 and 2, I see DHCPREQUESTS for both:
```
dnsmasq-dhcp[23680]: DHCPREQUEST(virbr1) 10.0.2.15 52:54:00:af:91:70
dnsmasq-dhcp[23680]: DHCPNAK(virbr1) 10.0.2.15 52:54:00:af:91:70 wrong network
dnsmasq-dhcp[23717]: DHCPREQUEST(virbr2) 10.0.2.15 52:54:00:83:10:f1
dnsmasq-dhcp[23717]: DHCPNAK(virbr2) 10.0.2.15 52:54:00:83:10:f1 wrong network
dnsmasq-dhcp[23680]: DHCPDISCOVER(virbr1) 52:54:00:af:91:70
dnsmasq-dhcp[23680]: DHCPOFFER(virbr1) 192.168.123.140 52:54:00:af:91:70
dnsmasq-dhcp[23680]: DHCPREQUEST(virbr1) 192.168.123.140 52:54:00:af:91:70
dnsmasq-dhcp[23680]: DHCPACK(virbr1) 192.168.123.140 52:54:00:af:91:70 install
dnsmasq-dhcp[23717]: DHCPDISCOVER(virbr2) 52:54:00:83:10:f1
dnsmasq-dhcp[23717]: DHCPOFFER(virbr2) 192.168.124.221 52:54:00:83:10:f1
dnsmasq-dhcp[23717]: DHCPREQUEST(virbr2) 192.168.124.221 52:54:00:83:10:f1
dnsmasq-dhcp[23717]: DHCPACK(virbr2) 192.168.124.221 52:54:00:83:10:f1 install
```

So I hit two problems:
1. When I use the same network, the VMs knock each other's IPs out so only one survives.
2. When I use unique networks for each VM, something is preventing one of them from successfully sending out a DHCPREQUEST.

I'd like to not have to define unique networks at all, but it seems like something is restarting dnsmasq or reloading it in the sense that the IP leased to the first machine is lost. When I use multiple networks, it seems like somehow one of them is blocked or fails to allow the DHCP exchange to happen.

**To Reproduce**
1. Make two `Vagrantfiles` without networking items defined (e.g. use the `vagrant-libvirt` network)
2. Run `vagrant up` on both files

**Expected behavior**

Each VM made from `vagrant up` receives an IP and keeps it.

**Screenshots**

**Versions (please complete the following information):**:
- Libvirt version: 7.0.0-3 (libvirt-daemon)
- Vagrant version [output of `vagrant version`]: 2.2.19
- Vagrant plugins versions (including vagrant-libvirt) [output of `vagrant plugin list`]: `vagrant-libvirt (0.9.0, global)`

**Debug Log**
Attach Output of `VAGRANT_LOG=debug vagrant ... --provider=libvirt >vagrant.log 2>&1`

Debug logs from parallel builds:
- Kubernetes.box which successfully got an IP: [kubernetes.box.log](https://github.com/vagrant-libvirt/vagrant-libvirt/files/9285034/kubernetes.box.log)
- Storage-ceph.box which failed to get an IP: [storage-ceph.box.log](https://github.com/vagrant-libvirt/vagrant-libvirt/files/9284648/storage-ceph.box.log)

**A Vagrantfile to reproduce the issue:**

This is the `Vagrantfile` that `packer` uses, it uses GoLang to template all the `{{ }}` items.
```ruby
Vagrant.configure("2") do |config|

  config.vagrant.plugins = "vagrant-libvirt"

  config.vm.define 'source', autostart: false do |source|
    source.vm.box = '{{ .SourceBox }}'
  end
  config.vm.define 'output' do |output|
    output.vm.box = '{{ .BoxName }}'
    output.vm.box_url = 'file://{{ .BoxName }}'
  end

  config.ssh.username = 'root'
  config.ssh.password = '<redacted>'
  config.ssh.insert_key = '{{ .InsertKey }}'

  # These options assist the build, but a user may want to remove them.
  config.vm.allow_fstab_modification = false
  config.vm.allow_hosts_modification = false
  config.vm.synced_folder '.', '/vagrant', disabled: true

  # Configure Vagrant's credentials (Packer does not use these values).
  config.vm.provider :libvirt do |domain|
    domain.autostart = true
    domain.cpus = 32
    domain.driver = 'kvm'
    domain.memory = 8192

    # Size in GB (needs integer). This will not automatically resize the root, that needs to be done
    # by hand.
    domain.machine_virtual_size = 42

    # For parallel builds to work, they need their own networks. Otherwise libvirt recreates the network
    # and one of the builds will loose connectivity and fail.
    domain.management_network_autostart = true
    domain.management_network_keep = true
    domain.management_network_name = '{{ .BoxName }}'
    domain.nested = true
  end
end
```

Here's a simplified `Vagrantfile` that can be used to recreate the issue:
```ruby
Vagrant.configure("2") do |config|

  config.vagrant.plugins = "vagrant-libvirt"

  config.vm.define 'output' do |output|
    output.vm.box = 'ncn-common.box'
    output.vm.box_url = 'file://ncn-common.box'
  end

  config.ssh.username = 'root'
  config.ssh.password = '<redacted>'
  config.ssh.insert_key = true


  # These options assist the build, but a user may want to remove them.
  config.vm.allow_fstab_modification = false
  config.vm.allow_hosts_modification = false
  config.vm.synced_folder '.', '/vagrant', disabled: true

  # Configure Vagrant's credentials (Packer does not use these values).
  config.vm.provider :libvirt do |domain|
    domain.autostart = true
    domain.cpus = 32
    domain.driver = 'kvm'
    domain.memory = 8192


    # Size in GB (needs integer). This will not automatically resize the root, that needs to be done
    # by hand.
    domain.machine_virtual_size = 42

    # For parallel builds to work, they need their own networks. Otherwise libvirt recreates the network
    # and one of the builds will loose connectivity and fail.
    domain.management_network_autostart = true
    domain.management_network_keep = true
    domain.nested = true
  end
end
```

At this point I'm at a loss of what I can debug, the multiple network idea seemed like it was going to work until it flaked out in builds 3 and 4.

**Comment by rustydb** [comment]
I'm working on getting debug logs, I had packer debug logs but when I tried getting vagrant debug logs my build failed on some fluke. I'm working on a rebuild to get the vagrant debug log output for the k8s and ceph layers.

**Comment by rustydb** [comment]
Just had another build with unique networks defined where one of the builds never sent the DHCPREQUEST

(from `sudo journalctl -xeu libvirtd -f | grep virbr`)
```bash
dnsmasq-dhcp[23706]: DHCP, sockets bound exclusively to interface virbr1
dnsmasq-dhcp[23750]: DHCP, sockets bound exclusively to interface virbr2
dnsmasq-dhcp[23794]: DHCP, sockets bound exclusively to interface virbr3
dnsmasq-dhcp[23838]: DHCP, sockets bound exclusively to interface virbr4
dnsmasq-dhcp[23750]: DHCPREQUEST(virbr2) 10.0.2.15 52:54:00:44:18:4d
dnsmasq-dhcp[23750]: DHCPNAK(virbr2) 10.0.2.15 52:54:00:44:18:4d wrong network
dnsmasq-dhcp[23706]: DHCPREQUEST(virbr1) 10.0.2.15 52:54:00:44:18:4d
dnsmasq-dhcp[23706]: DHCPNAK(virbr1) 10.0.2.15 52:54:00:44:18:4d wrong network
dnsmasq-dhcp[23750]: DHCPDISCOVER(virbr2) 52:54:00:44:18:4d
dnsmasq-dhcp[23750]: DHCPOFFER(virbr2) 192.168.124.201 52:54:00:44:18:4d
dnsmasq-dhcp[23750]: DHCPREQUEST(virbr2) 192.168.124.201 52:54:00:44:18:4d
dnsmasq-dhcp[23750]: DHCPACK(virbr2) 192.168.124.201 52:54:00:44:18:4d install
dnsmasq-dhcp[23706]: DHCPDISCOVER(virbr1) 52:54:00:44:18:4d
dnsmasq-dhcp[23706]: DHCPOFFER(virbr1) 192.168.123.201 52:54:00:44:18:4d
dnsmasq-dhcp[23706]: DHCPREQUEST(virbr1) 192.168.123.201 52:54:00:44:18:4d
dnsmasq-dhcp[23706]: DHCPACK(virbr1) 192.168.123.201 52:54:00:44:18:4d install
dnsmasq-dhcp[23794]: DHCPREQUEST(virbr3) 192.168.123.201 52:54:00:44:18:4d
dnsmasq-dhcp[23794]: DHCPNAK(virbr3) 192.168.123.201 52:54:00:44:18:4d wrong network
dnsmasq-dhcp[23794]: DHCPDISCOVER(virbr3) 52:54:00:44:18:4d
dnsmasq-dhcp[23794]: DHCPOFFER(virbr3) 192.168.125.201 52:54:00:44:18:4d
dnsmasq-dhcp[23794]: DHCPREQUEST(virbr3) 192.168.125.201 52:54:00:44:18:4d
dnsmasq-dhcp[23794]: DHCPACK(virbr3) 192.168.125.201 52:54:00:44:18:4d ncn
```

```bash
$ sudo virsh net-list --all
 Name               State      Autostart   Persistent
-------------------------------------------------------
 default            inactive   no          yes
 kubernetes.box     active     yes         yes
 ncn-common.box     active     yes         yes
 pit-common.box     active     yes         yes
 storage-ceph.box   active     yes         yes
```

```bash
jenkins@csm-metal-builder-large-2w4jj8:~$ sudo virsh net-dumpxml kubernetes.box
<network>
  <name>kubernetes.box</name>
  <uuid>39a7f396-8c29-4d2b-ace1-8ecb76b5a521</uuid>
  <forward mode='nat'>
    <nat>
      <port start='1024' end='65535'/>
    </nat>
  </forward>
  <bridge name='virbr3' stp='on' delay='0'/>
  <mac address='52:54:00:88:34:d4'/>
  <ip address='192.168.125.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.125.2' end='192.168.125.254'/>
    </dhcp>
  </ip>
</network>

jenkins@csm-metal-builder-large-2w4jj8:~$ sudo virsh net-dumpxml storage-ceph.box
<network connections='1'>
  <name>storage-ceph.box</name>
  <uuid>2d928f53-8394-4554-a57e-475b6a10fc5a</uuid>
  <forward mode='nat'>
    <nat>
      <port start='1024' end='65535'/>
    </nat>
  </forward>
  <bridge name='virbr4' stp='on' delay='0'/>
  <mac address='52:54:00:25:b6:7b'/>
  <ip address='192.168.126.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.126.2' end='192.168.126.254'/>
    </dhcp>
  </ip>
</network>
```

**Comment by electrofelix** [comment]
@rustydb while I can't be sure I can explain everything you are seeing, typically these issues are all caused by stray information from when the VM is initially packaged containing specific network definitions that aren't removed before shutdown.

I think the machines colliding when on the same network can be partially traced to a duplicate machine-id issue similar to https://kb.vmware.com/s/article/82229, so solving should just involve deleting the file as a clean up step in packer.

It doesn't fully explain everything though, as usually stray network information usually results in a consistent failure to request a DHCP address, and it doesn't sound consistent here. 

So for the second issue, it might be the predictable device naming combined with old network information being left in the VM before packer packages it up. You'll need to wipe the netplan information or disable the systems predicable device naming or potentially both. Used to be just wipe the network device from the use rules before packaging. Recently somethings have changed and I'm not sure what's needed for suse based distros these days.

I'll see if I can find a know working box somewhere using suse leap and the packaging repo to build it might give a few hints. But could a couple of days before I can dig up the correct cleanup recipes.

Hopefully these might help you find the solutions in the meantime.

**Comment by rustydb** [comment]
@electrofelix I'll give purging the machine-id a try.

This is perplexing, but perhaps related to the machine-id, but dnsmasq seems to offer the already leased IP to the new MAC. in this output below, I ran two boxes using the same network after disabling these two options and repackaging the box
```bash
sed -i 's/DHCLIENT_USE_LAST_LEASE="yes"/DHCLIENT_USE_LAST_LEASE="no"/g' /etc/sysconfig/network/dhcp
sed -i 's/DHCLIENT_HOSTNAME_OPTION="AUTO"/DHCLIENT_HOSTNAME_OPTION=""/g' /etc/sysconfig/network/dhcp
```

```bash
DHCPDISCOVER(virbr5) 52:54:00:84:a1:94
DHCPOFFER(virbr5) 192.168.121.254 52:54:00:84:a1:94
DHCPREQUEST(virbr5) 192.168.121.254 52:54:00:84:a1:94
DHCPACK(virbr5) 192.168.121.254 52:54:00:84:a1:94
DHCPDISCOVER(virbr5) 52:54:00:36:aa:d7
DHCPOFFER(virbr5) 192.168.121.254 52:54:00:36:aa:d7
DHCPREQUEST(virbr5) 192.168.121.254 52:54:00:36:aa:d7
DHCPACK(virbr5) 192.168.121.254 52:54:00:36:aa:d7
```

**Comment by rustydb** [comment]
@electrofelix in my base box I do purge the machine-id and some udev rules:
```bash
#!/usr/bin/env bash

set -ex

echo "removing our autoyast cache to ensure no lingering sensitive content remains there from install"
rm -rf /var/adm/autoinstall/cache

zypper clean --all

echo "clean up network interface persistence"
rm -f /etc/udev/rules.d/70-persistent-net.rules;
touch /etc/udev/rules.d/75-persistent-net-generator.rules;

echo "truncate any logs that have built up during the install"
find /var/log/ -type f -name "*.log.*" -exec rm -rf {} \;
find /var/log -type f -exec truncate --size=0 {} \;

echo "remove the contents of /tmp and /var/tmp"
rm -rf /tmp/* /var/tmp/*

echo "blank netplan machine-id (DUID) so machines get unique ID generated on boot"
truncate -s 0 /etc/machine-id

echo "force a new random seed to be generated"
rm -f /var/lib/systemd/random-seed

echo "clear the history so our install isn't there"
rm -f /root/.wget-hsts
export HISTSIZE=0
```

but I don't do that again at the end of my layer 2 boxes, so perhaps that's why kubernetes and storage-ceph struggle but pit-common and ncn-common don't..  I'll give that a whirl and post back.

**Comment by rustydb** [comment]
If it would be helpful, I can package our base box up with a different password (e.g. `vagrant`) for sharing here.

**Comment by rustydb** [comment]
Closing this.

Instead of taking our qemu sles15-base and converting it to a box with a packer post-provioner, I fixed the issue by using a vagrant builder that loads the upsteam SUSE SLES box. Something was weird with converting the other builders' artifacts to Vagrant, I'm not getting the parallel issue when I spin up multiple boxes anymore.

**Comment by electrofelix** [comment]
I'm glad you have a solution to your problem. I still haven't had a chance to have a proper look. I suspect some of the issues are to do with the other providers don't use a network connection to manage the VM and can access it directly.

I'm hopeful that we might be able to achieve something similar in the near future building on some initial usage of qemu-agent, combined with some changes with how port forwarding is done to get ssh access without a network. This would solve your problem as well I suspect as the behaviour would then be much closer to the other providers as well.

**Comment by rustydb** [comment]
@electrofelix actually, now I figured it out.

I thought I had fixed the issue by using the upstream SLES-15-SP3 box made by SUSE: https://build.opensuse.org/package/show/Virtualization:Appliances:Images:SLE-15-SP3/SLES-15-SP3-Vagrant
When I used that box, I could run parallel builds. However the boxes I produced still couldn't run in parallel..

To me this meant that something we're doing (or not doing) in our pipeline is preventing parallel builds from working since the upstream box worked. However even when I ran the base box and then saved it right away (without any changes) to a new box file, that new box file would fail to run parallel builds.

I ended up discovering that `wickedd` was storing ID information in `/var/lib/wicked` that needed to be purged before creating a box file. After purging `/var/lib/wicked/*.xml` my packaged boxes were able to run parallel builds.

I discovered this by booting the base SUSE box, and running `find /var -mtime -1` to see what files had changed during my session. I also ran it against `/etc` and several other dirs, but only `/etc` and `/var` had anything interesting (most other directories had 0 changes).

```bash
localhost:/ # find var -mtime -1
var
var/log/wtmp
var/log/lastlog
var/lib/wicked
var/lib/wicked/duid.xml
var/lib/wicked/lease-eth0-dhcp-ipv4.xml
var/lib/wicked/iaid.xml
var/lib/systemd
var/lib/systemd/timers
var/lib/systemd/timers/stamp-fstrim.timer
var/lib/systemd/linger
var/lib/systemd/random-seed
var/lib/systemd/catalog
var/lib/systemd/catalog/database
var/tmp
var/tmp/systemd-private-8962b219f01b44fbbe46dcceff95530e-systemd-logind.service-5E1Rkf
var/tmp/systemd-private-8962b219f01b44fbbe46dcceff95530e-systemd-logind.service-5E1Rkf/tmp
localhost:/ # find etc -mtime -1
etc
etc/fstab
etc/resolv.conf
etc/udev
etc/udev/hwdb.bin
etc/udev/rules.d
etc/udev/rules.d/70-persistent-net.rules
etc/ssh
etc/ssh/ssh_host_rsa_key
etc/ssh/ssh_host_ecdsa_key.pub
etc/ssh/ssh_host_ecdsa_key
etc/ssh/ssh_host_ed25519_key
etc/ssh/ssh_host_dsa_key
etc/ssh/ssh_host_ed25519_key.pub
etc/ssh/ssh_host_rsa_key.pub
etc/ssh/ssh_host_dsa_key.pub
etc/machine-id
```

We were already truncating `/etc/machine-id`, so it wasn't that... but I noticed the wicked files:
```xml
<duid>00:01:00:01:2a:9a:60:09:52:54:00:b2:0d:d0</duid>
<iaid device="eth0">11668944</iaid>
<lease>
  <family>ipv4</family>
  <type>dhcp</type>
  <uuid>89a30763-ca74-0e00-0902-000004000000</uuid>
  <state>granted</state>
  <acquired>1661445004</acquired>
  <update>0x00000000</update>
  <ipv4:dhcp>
    <client-id>ff:00:b2:0d:d0:00:01:00:01:2a:9a:60:09:52:54:00:b2:0d:d0</client-id>
    <server-id>192.168.121.1</server-id>
    <sender-hw-address>52:54:00:9f:c4:42</sender-hw-address>
    <lease-time>3600</lease-time>
    <renewal-time>1800</renewal-time>
    <rebind-time>3150</rebind-time>
    <address>192.168.121.185</address>
    <netmask>255.255.255.0</netmask>
    <broadcast>192.168.121.255</broadcast>
    <boot>
      <server-address>192.168.121.1</server-address>
    </boot>
    <routes>
      <route>
        <nexthop>
          <gateway>192.168.121.1</gateway>
        </nexthop>
      </route>
    </routes>
    <dns>
      <server>192.168.121.1</server>
    </dns>
  </ipv4:dhcp>
</lease>
```

The important ones seemed to be the `duid.xml` and `iaid.xml`. After adding `rm -f /var/lib/wicked/*.xml` to my cleanup provisioner script I stopped running into this problem.

I've retitled this issue to include "SUSE" in the name to help others that hit his problem.

tl;dr your original assessment was right, after looking through the booted SUSE box I discovered left-over IP/NIC information from `wicked`.

**Comment by electrofelix** [comment]
@rustydb thanks for following up on that, not only am I happy your problem is fixed, it's great to finally know how to deal with this for any SUSE image.

I'm hopeful that at some point in the future the management network will be deprecated and unnecessary, side stepping the whole area of needing any network state to be correctly cleaned up in the VM before packaging.
